components/PocPrompt.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Text, View, TextInput, TouchableWithoutFeedback, Button, StyleSheet
+} from 'react-native';
+
+const styles = StyleSheet.create({
+  container: {
+    flexDirection: 'column',
+    justifyContent: 'flex-start',
+    alignItems: 'center'
+  }
+});
+
+// TODO: Impliment api request
+const apiGetRequest = async () => 1500;
+
+// TODO: Impliment api request
+// eslint-disable-next-line no-unused-vars
+const apiPutRequest = async (calorieGoal: number) => { };
+
+export default function PocPrompt() {
+  // State
+  const [displayPrompt, setDisplayPrompt] = useState(false);
+  const [calorieGoal, setCalorieGoal] = useState<number>(undefined);
+
+  // Local Functions
+  const toggleDisplayPrompt = () => {
+    setDisplayPrompt(!displayPrompt);
+  };
+
+  const getGoal = async () => {
+    const goal = await apiGetRequest();
+    setCalorieGoal(goal);
+  };
+
+  const changeGoal = async () => {
+    await apiPutRequest(calorieGoal);
+    setDisplayPrompt(false);
+  };
+
+  // Fetch the initial calorie goal on load
+  useEffect(() => {
+    getGoal();
+  }, []);
+
+  // Render
+  if (displayPrompt) {
```

Comment:
There seems to be some duplication between these two cases. Would it be possible to render parts of the component differently based on state rather than having two separately defined layouts?

Reply:
If you're referring to the <View> tags wrapping each view, I don't think so because JSX requires a single parent element. So the view tags need to be present on both pieces of JSX individually.